modules/social_features/social_group/social_group.module

```diff
@@ -726,6 +726,18 @@ function social_group_form_alter(&$form, FormStateInterface $form_state, $form_i
       array_unshift($form['actions']['submit']['#submit'], '_social_group_delete_group');
     }
   }
+
+  if (in_array($form_id, ['group_flexible_group_edit_form', 'group_flexible_group_add_form'])) {
+    $join_method_default_value = 'added';
+    if (in_array('direct',  $form['field_group_allowed_join_method']['widget']['#default_value'])) {
+      $join_method_default_value = 'direct';
+    }
+    else if (in_array('request',  $form['field_group_allowed_join_method']['widget']['#default_value'])) {
+      $join_method_default_value = 'request';
+    }
+    $form['field_group_allowed_join_method']['widget']['#type'] = 'radios';
+    $form['field_group_allowed_join_method']['widget']['#default_value'] = $join_method_default_value;
+  }
```

Comment:
I think it should be in the **Social Group Request** module because this module provides new join methods

Reply:
It should be there because changing widget type to radios should work regardless social_group_request feature.